### ISSUE TITLE
ember init failing to setup projects in an empty folder

### ISSUE THREAD

**Issue by jrm2k6** [ember-cli/ember-cli#4213]
```
node -v : v0.12.4
npm -v 2.10.1
ember-cli 0.2.7
watchman -v 3.0.0
```

command: ```ember init ```

error:

```
± % ember init                                                                                                                                                        
version: 0.2.7
Cannot read property 'toLowerCase' of undefined
TypeError: Cannot read property 'toLowerCase' of undefined
    at module.exports (/usr/local/lib/node_modules/ember-cli/lib/utilities/valid-project-name.js:4:14)
    at Class.module.exports.Command.extend.run (/usr/local/lib/node_modules/ember-cli/lib/commands/init.js:71:10)
    at Class.<anonymous> (/usr/local/lib/node_modules/ember-cli/lib/models/command.js:136:17)
    at lib$rsvp$$internal$$tryCatch (/usr/local/lib/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:489:16)
    at lib$rsvp$$internal$$invokeCallback (/usr/local/lib/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:501:17)
    at lib$rsvp$$internal$$publish (/usr/local/lib/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:472:11)
    at lib$rsvp$asap$$flush (/usr/local/lib/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:1290:9)
    at process._tickCallback (node.js:355:11)
```

I am trying to setup an ember app in the public directory of a Laravel app.

Using ember new name-app works but it doesn't create the ember structure in the directory from which I run the command, but goes up to find a folder containing a package.json which I don't want.

**Comment by stefanpenner** [comment]
which watchman version?

**Comment by stefanpenner** [comment]
in what folder are you when you run `ember init` ?

**Comment by jrm2k6** [comment]
1) 3.0.0 
2) I run it from public/assets/ember. ember is an newly created folder. assets contains some other folders. public contains a package.json.
3) MacOSX Yosemite 10.10.3

**Comment by stefanpenner** [comment]
is there a package.json somehwere up the hierarchy ?

**Comment by jrm2k6** [comment]
Yep, two levels up.

**Comment by stefanpenner** [comment]
what are its contents?

**Comment by jrm2k6** [comment]
```
{
  "dependencies": {
    "browser-sync": "^2.0.0-rc1",
    "browserify": "^8.1.3",
    "envify": "^3.4.0",
    "es6-shim": "^0.26.1",
    "fixed-data-table": "^0.1.0",
    "glob": "^4.3.5",
    "gulp": "^3.8.10",
    "gulp-autoprefixer": "^2.1.0",
    "gulp-coffee": "^2.2.0",
    "gulp-if": "^1.2.5",
    "gulp-notify": "^2.2.0",
    "gulp-rename": "^1.2.0",
    "gulp-ruby-sass": "^1.0.0-alpha.2",
    "gulp-size": "^1.2.0",
    "gulp-uglify": "^1.1.0",
    "gulp-uncss": "^1.0.0",
    "gulp-util": "^3.0.2",
    "markdown": "^0.5.0",
    "moment": "^2.10.2",
    "object-assign": "^2.0.0",
    "react": "^0.12.2",
    "react-loader": "^1.2.0",
    "react-markdown-editor": "0.0.5",
    "react-modal": "^0.1.0",
    "react-router": "^0.12.0",
    "reactify": "^1.0.0",
    "reflux": "^0.2.5",
    "run-sequence": "^0.3.2",
    "vinyl-source-stream": "^1.0.0"
  },
  "devDependencies": {
    "gulp-debug": "^2.0.0"
  }
}
```

**Comment by stefanpenner** [comment]
i'll need the whole thing

**Comment by jrm2k6** [comment]
I guess we have our solution then. The package.json is only listing dependencies, nothing else. So it might be messed up because of this. Is init looking for something specific in package.json?

**Comment by jrm2k6** [comment]
It does create something, but not in the current folder. I said no when asking to override package.json

**Comment by stefanpenner** [comment]
we should make our code more robust, so that it handles this scenario.

**Comment by jrm2k6** [comment]
Can you guide me to the part of the code doing it and I can check it tonight, or tomorrow?

**Comment by jrm2k6** [comment]
How does the override works in few words?

**Comment by stefanpenner** [comment]
> at module.exports (/usr/local/lib/node_modules/ember-cli/lib/utilities/valid-project-name.js:4:14)

the stack trace should be a good hint

**Comment by jrm2k6** [comment]
It is. Was just looking for some more information if you had any from the top of your head but anyway.

**Comment by stefanpenner** [comment]
maybe somehwere around https://github.com/ember-cli/ember-cli/blob/master/lib/commands/init.js#L62

**Comment by jrm2k6** [comment]
So, before I read some more code.

ember new name-project will look for a folder containing a package.json and create the ember app in the folder.

ember init is supposed to create a project whether or not there is a package.json in the current folder, but grab some info for a package.json it found somewhere in the hierarchy.

Right now, ember new and ember init do the same thing?

**Comment by dukex** [comment]
Hi @jrm2k6 and @stefanpenner 

This issues was fixed by #4270

**Comment by jrm2k6** [comment]
Great! Thank you!
